examples/regression/plot_timeseries_enbpi.py

```diff
@@ -0,0 +1,143 @@
+"""
+==================================================================
+Estimating prediction intervals of time series forecast with EnbPI
+==================================================================
+This example uses
+:class:`mapie.time_series_regression.MapieTimeSeriesRegressor` to estimate
+prediction intervals associated with time series forecast. The implementation
+is still at its first step, based on Jackknife+-after-bootsrtap, to estimate
+residuals and associated prediction intervals.
+
+We use here the Victoria electricity demand dataset used in the book
+"Forecasting: Principles and Practice" by R. J. Hyndman and G. Athanasopoulos.
+The electricity demand features daily and weekly seasonalities and is impacted
+by the temperature, considered here as a exogeneous variable.
+
+The data is modelled by a Random Forest model with a
+:class:`sklearn.model_selection.RandomizedSearchCV` using a sequential
+:class:`sklearn.model_selection.TimeSeriesSplit` cross validation, in which the
+training set is prior to the validation set.
+The best model is then feeded into
+:class:`mapie.time_series_regression.MapieTimeSeriesRegressor` to estimate the
+associated prediction intervals. We compare two approaches: one with no
+`partial_fit` call and one with `partial_fit` every 5 steps.
+"""
+import warnings
+
+import numpy as np
+import pandas as pd
+from matplotlib import pylab as plt
+from sklearn.ensemble import RandomForestRegressor
+
+from mapie.metrics import regression_coverage_score
+from mapie.subsample import Subsample
+from mapie.time_series_regression import MapieTimeSeriesRegressor
+
+warnings.simplefilter("ignore")
+
+# Load input data and feature engineering
+demand_df = pd.read_csv(
+    "../data/demand_temperature.csv", parse_dates=True, index_col=0
+)
+demand_df["Date"] = pd.to_datetime(demand_df.index)
+demand_df["Weekofyear"] = demand_df.Date.dt.isocalendar().week.astype("int64")
+demand_df["Weekday"] = demand_df.Date.dt.isocalendar().day.astype("int64")
+demand_df["Hour"] = demand_df.index.hour
+for hour in range(1, 5):
+    demand_df[f"Lag_{hour}"] = demand_df["Demand"].shift(hour)
+
+# Train/validation/test split
+num_test_steps = 24 * 7 * 2
+demand_train = demand_df.iloc[:-num_test_steps, :].copy()
+demand_test = demand_df.iloc[-num_test_steps:, :].copy()
+features = ["Weekofyear", "Weekday", "Hour", "Temperature"] + [
+    f"Lag_{hour}" for hour in range(1, 5)
+]
+X_train = demand_train.loc[
+    ~np.any(demand_train[features].isnull(), axis=1), features
+]
+y_train = demand_train.loc[X_train.index, "Demand"]
+X_test = demand_test.loc[:, features]
+y_test = demand_test["Demand"]
+
+# Model
+model = RandomForestRegressor(max_depth=15, n_estimators=673, random_state=59)
```

Comment:
Update the description by saying that `RandomForestRegressor` has been optimized before running the script ?

Reply:
Done